synapse/lib/atomfile.py

```diff
@@ -80,8 +92,6 @@ def writeoff(self, off, byts):
         Atomically write bytes at the given offset.
         '''
         bsize = len(byts)
-        if off + bsize > self.size:
```

Comment:
This check was removed from `def writeoff` and moved into a single AtomFile's implementation of `_writeoff`.  Are we making the conscious decision for let some types of atomfiles blow up?  That feels unsafe.

Reply:
Actually, many types of atomfile instances will not have to blow up ( because they can resize to larger with no effort ).